src/ripple/app/tx/impl/OfferStream.h

```diff
@@ -174,6 +174,16 @@ class FlowOfferStream : public TOfferStreamBase<TIn, TOut>
 public:
     using TOfferStreamBase<TIn, TOut>::TOfferStreamBase;
 
+    // The following interface allows offer crossing to permanently
+    // remove self crossed offers.  The motivation is somewhat
+    // unintuitive.  See the discussion in the testSelfCrossOffer1()
+    // unit test.
```

Comment:
I hear that.  I put the description on the unit test for 2 reasons:

1. The unit test was how I discovered the oddness and demonstrates the desired behavior.
2. The code the accomplishes the desired behavior is distributed into three places.  Attaching the description to a dedicated test, and referencing that description from three places, reduced my inclination toward duplicating the description in three places.

But I can reorganize it if you'd like.